src/components/CurrencySelector.tsx

```diff
@@ -0,0 +1,116 @@
+import { Payment } from './PaymentCard';
+import { NetworkInfo, CryptoAsset } from '@windingtree/win-commons/dist/types';
+import { useState, useMemo, useCallback, useEffect } from 'react';
+import {
+  FormControl,
+  FormLabel,
+  RadioGroup,
+  FormControlLabel,
+  Radio
+} from '@mui/material';
+import { BigNumber, utils } from 'ethers';
+import { getNetworkInfo } from '../config';
+import Logger from '../utils/logger';
+
+const logger = Logger('CurrencySelector');
+
+export interface CheckoutPrice {
+  value: BigNumber;
+  currency: string;
+}
+
+export interface PriceSelectProps {
+  payment?: Payment;
+  network?: NetworkInfo;
+  onQuote(useQuote: boolean): void;
+}
+
+export const CurrencySelector = ({ payment, network, onQuote }: PriceSelectProps) => {
+  const [value, setValue] = useState<string | null>(payment ? payment.currency : null);
+
+  const options = useMemo<CheckoutPrice[]>(
+    () =>
+      payment
+        ? [
+            {
+              value: payment.value,
+              currency: payment.currency
+            },
+            ...(payment.quote
+              ? [
+                  {
+                    value: BigNumber.from(
+                      utils.parseEther(payment.quote.targetAmount ?? '0')
+                    ),
+                    currency: payment.quote.targetCurrency ?? 'USD'
+                  }
+                ]
+              : [])
+          ]
+        : [],
+    [payment]
+  );
+
+  const tokens = useMemo<CryptoAsset[]>(() => {
+    if (!payment || !network) {
+      return [];
+    }
+    const chain = getNetworkInfo(network.chainId);
+    return [...chain.contracts.assets.filter((a) => a.currency === payment.currency)];
+  }, [network, payment]);
+
+  const handleChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const newValue = (event.target as HTMLInputElement).value;
+      setValue(newValue);
+      logger.debug('Currency selected:', newValue);
+      const nextPrice = options.find((o) => o.currency === newValue) ?? null;
+      const useQuote = nextPrice?.currency === 'USD';
```

Comment:
I would personally not use `useQuote` and instead `quote`, as `use` suggest it is a `hook`.

Reply:
makes sense

Reply:
renamed to the `withQuote`